libsmtutil/CHCSmtLib2Interface.cpp

```diff
@@ -198,242 +180,195 @@ std::string CHCSmtLib2Interface::querySolver(std::string const& _input)
+			{
+				SortPointer sort;
+				std::vector<smtutil::Expression> arguments;
+				if (isAtom(_subExpr.front()))
 				{
-					SortPointer sort;
-					std::vector<smtutil::Expression> arguments;
-					if (isAtom(_subExpr.front()))
+					std::string const& op = asAtom(_subExpr.front());
+					if (op == "!")
 					{
-						std::string const& op = asAtom(_subExpr.front());
-						if (op == "!")
-						{
-							// named term, we ignore the name
-							smtAssert(_subExpr.size() > 2);
-							return toSMTUtilExpression(_subExpr[1]);
-						}
-						if (op == "exists" || op == "forall")
-						{
-							smtAssert(_subExpr.size() == 3);
-							smtAssert(!isAtom(_subExpr[1]));
-							return parseQuantifier(op, asSubExpressions(_subExpr[1]), _subExpr[2]);
-						}
-						for (size_t i = 1; i < _subExpr.size(); i++)
-							arguments.emplace_back(toSMTUtilExpression(_subExpr[i]));
-						if (auto tupleSort = lookupKnownTupleSort(op); tupleSort)
-						{
-							auto sortSort = std::make_shared<SortSort>(tupleSort.value());
-							return Expression::tuple_constructor(Expression(sortSort), arguments);
-						}
-						if (auto it = m_knownVariables.find(op); it != m_knownVariables.end())
-							return smtutil::Expression(op, std::move(arguments), it->second);
-						else
-						{
-							std::set<std::string>
-								boolOperators{"and", "or", "not", "=", "<", ">", "<=", ">=", "=>"};
-							sort = contains(boolOperators, op) ? SortProvider::boolSort : arguments.back().sort;
-							return smtutil::Expression(op, std::move(arguments), std::move(sort));
-						}
-						smtAssert(false, "Unhandled case in expression conversion");
+						// named term, we ignore the name
+						smtAssert(_subExpr.size() > 2);
+						return toSMTUtilExpression(_subExpr[1]);
+					}
+					if (op == "exists" || op == "forall")
+					{
+						smtAssert(_subExpr.size() == 3);
+						smtAssert(!isAtom(_subExpr[1]));
+						return parseQuantifier(op, asSubExpressions(_subExpr[1]), _subExpr[2]);
+					}
+					for (size_t i = 1; i < _subExpr.size(); i++)
+						arguments.emplace_back(toSMTUtilExpression(_subExpr[i]));
+					if (auto tupleSort = lookupKnownTupleSort(op); tupleSort)
+					{
+						auto sortSort = std::make_shared<SortSort>(tupleSort.value());
+						return Expression::tuple_constructor(Expression(sortSort), arguments);
+					}
+					if (m_context.isDeclared(op))
+						return smtutil::Expression(op, std::move(arguments), m_context.getDeclaredSort(op));
+					if (auto maybeTupleAccessor = m_context.getTupleAccessor(op); maybeTupleAccessor.has_value())
+					{
+						auto accessor = maybeTupleAccessor.value();
+						return smtutil::Expression("dt_accessor_" + accessor.first, std::move(arguments), accessor.second);
 					}
 					else
 					{
-						// check for const array
-						if (_subExpr.size() == 2 and !isAtom(_subExpr[0]))
+						std::set<std::string>
+							boolOperators{"and", "or", "not", "=", "<", ">", "<=", ">=", "=>"};
```

Comment:
```suggestion
						std::set<std::string> boolOperators{
							"and",
							"or",
							"not",
							 "=",
							 "<",
							 ">",
							 "<=",
							 ">=",
							 "=>"
						};
```
I think this would be more in line with our coding style guide, although I think this could also be only one line albeit a bit long.